src/main/java/io/gravitee/resource/oauth2/generic/OAuth2GenericResource.java

```diff
@@ -267,11 +258,18 @@ public void handle(AsyncResult<HttpClientResponse> asyncResponse) {
                                                             responseHandler.handle(new OAuth2Response(true, content));
                                                         }
                                                     } catch (IOException e) {
-                                                        logger.error("Unable to validate introspection endpoint payload: {}", content);
-                                                        responseHandler.handle(new OAuth2Response(false, content));
+                                                        logger.error("Unable to validate introspection endpoint payload: {}", content, e);
+                                                        responseHandler.handle(new OAuth2Response(e));
                                                     }
                                                 } else {
-                                                    responseHandler.handle(new OAuth2Response(false, buffer.toString()));
+                                                    logger.error(
+                                                        "An error occurs while checking OAuth2 token. Request ends with status {}: {}",
+                                                        response.statusCode(),
+                                                        buffer.toString()
+                                                    );
+                                                    responseHandler.handle(
+                                                        new OAuth2Response(new Throwable("An error occurs while checking OAuth2 token"))
```

Comment:
That the limit I pointed out yesterday, I would prefer having an explicit exception here than a generic Throwable.

Reply:
Use now `OAuth2ResourceException`